File: src/main/java/io/reflectoring/cleantimetracker/projectcontext/domain/usecase/listtasks/ListTasksUseCase.java
package io.reflectoring.cleantimetracker.projectcontext.domain.usecase.listtasks;

import io.reflectoring.cleantimetracker.projectcontext.domain.entity.ProjectId;
import io.reflectoring.cleantimetracker.projectcontext.domain.entity.Task;
import io.reflectoring.cleantimetracker.projectcontext.domain.port.out.persistence.QueryTasksPort;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ListTasksUseCase {

	private QueryTasksPort queryTasksPort;

	public ListTasksUseCase(QueryTasksPort queryTasksPort) {
		this.queryTasksPort = queryTasksPort;
	}

	public List<Task> listTasksForProject(ProjectId projectId) {
		return queryTasksPort.listTasksForProject(projectId);
	}

}

File: src/main/java/io/reflectoring/cleantimetracker/projectcontext/domain/usecase/listtasks/ListTasksUseCase.kt
package io.reflectoring.cleantimetracker.projectcontext.domain.usecase.listtasks

import io.reflectoring.cleantimetracker.projectcontext.domain.entity.ProjectId
import io.reflectoring.cleantimetracker.projectcontext.domain.entity.Task
import io.reflectoring.cleantimetracker.projectcontext.domain.port.out.persistence.QueryTasksPort
import org.springframework.stereotype.Service

@Service
class ListTasksUseCase(private val queryTasksPort: QueryTasksPort) {
    fun listTasksForProject(projectId: ProjectId?): List<Task> {
        return queryTasksPort.listTasksForProject(projectId)
    }

}